### PR TITLE
Rust Code Generation

### PR DESCRIPTION
**Pull request by anuragmakineni**
Support Rust code generation in Symforce

### CODE REVIEW

**Review by gcross-zipline — COMMENTED** [review]
Seems reasonable to me - I didn't spot anything suspicious.

> Reviewed 1 of 9 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ZiplineTeam/symforce/3#-:-NVfxC418s5RyQBRZt91:bnocu3k)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @anuragmakineni)

File: symforce/codegen/type_description.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __init__(self,
+                 typename: str,
+                 unformatted_typenames: T.List[str],
+                 full_typename: str,
+                 index: T.Dict[str, IndexEntry]):
+        """
+        Construct new TypeDescription.
+        Args:
+            TODO(gareth): Fill this out.
```
Comment (gcross-zipline) [inline symforce/codegen/type_description.py]: Might as well delete this TODO, I don't have anything substantive to add here. Maybe add a comment on this object that it just replaces the dict that was previously built by `build_types_dict`

Comment (anuragmakineni) [reply symforce/codegen/type_description.py]: done!

File: symforce/codegen/backends/rust/rust_code_printer.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        return "Scalar::from_f64(std::f64::consts::E).unwrap()"
+
+    def _print_Pi(self, expr: sympy.core.numbers.Pi, _type=False) -> str:
+        """
+        Customizations:
+            * Reference rust standard library, and cast appropriately.
+        """
+        return "Scalar::from_f64(std::f64::consts::PI).unwrap()"
+
+    def _print_Infinity(self, expr, _type=False):
```
Comment (gcross-zipline) [inline symforce/codegen/backends/rust/rust_code_printer.py]: Missing type annotations (though it doesn't really matter much here, the type is fairly obvious)

**Review by gcross-zipline — APPROVED** [review]

File: symforce/codegen/backends/rust/rust_code_printer.py (hunk 2 above)
Comment (anuragmakineni) [reply symforce/codegen/backends/rust/rust_code_printer.py]: went in and got all these that were missing!

**Pull request merged**